build_resolvers/test/resolver_test.dart

```diff
@@ -355,6 +355,25 @@ void main() {
             AssetId('a', 'lib/b.dart'));
       }, resolvers: AnalyzerResolvers());
     });
+
+    test('Respects withEnabledExperiments', () async {
+      Logger.root.level = Level.ALL;
+      Logger.root.onRecord.listen(print);
+      await withEnabledExperiments(
+          () => resolveSources({
+                'a|web/main.dart': '''
+// @dart=${sdkLanguageVersion.major}.${sdkLanguageVersion.minor}
```

Comment:
1. What version do you get here?
2. What version is in `ExperimentStatus.currentVersion` in analyzer?
3. Do you have a version of analyzer that includes https://dart-review.googlesource.com/c/sdk/+/145220?

Reply:
> * What version do you get here?

2.9

> * What version is in `ExperimentStatus.currentVersion` in analyzer?

2.9

> * Do you have a version of analyzer that includes https://dart-review.googlesource.com/c/sdk/+/145220?

I have reverted to not call `restrictToVersion` at all if any experiments are enabled, so it shouldn't matter. Instead it now throws if you attempt to use an experiment and the analyzer and SDK don't agree on a language version.

Reply:
The test also has a check that the language version is correct, (it expects the library language version to equal the sdk language version)

Reply:
I'm not sure where you reverted the call to `restrictToVersion`, but analyzer [calls it itself](https://github.com/dart-lang/sdk/blob/2ccfc88f25e378ac82e2548bd8f5117d8ebea6ab/pkg/analyzer/lib/src/dart/analysis/feature_set_provider.dart#L33) to determine the set of enabled features for a file.

Reply:
My analyzer version though fwiw is the latest published version, 0.39.8. I don't see any tags in the SDK for this release so I don't know if its included or not.

Reply:
Try to set breakpoint in `FeatureSetProvider` and check which versions and features you get.

Reply:
ok, thanks for that code pointer @scheglov I noticed there that for non-packages there is a totally different feature set that has to be set, the `nonPackageFeatureSet`. If I set that in addition to the `contextFeatures` then the test passes 🎉

Reply:
Actually, it looks like it also requires a dependency override on analyzer/_fe_analyzer_shared to the latest version in the SDK (to pick up that fix).

When can we expect a new version of analyzer to be published?